influxql/ast.go

```diff
@@ -1036,34 +1034,70 @@ func (s *SelectStatement) RewriteTimeFields() {
 // ColumnNames will walk all fields and functions and return the appropriate field names for the select statement
 // while maintaining order of the field names
 func (s *SelectStatement) ColumnNames() []string {
-	columnNames := []string{}
-
-	if !s.OmitTime {
-		columnNames = append(columnNames, "time")
-	}
-
-	// First walk each field
+	// First walk each field to determine the number of columns.
+	columnFields := Fields{}
 	for _, field := range s.Fields {
+		columnFields = append(columnFields, field)
+
 		switch f := field.Expr.(type) {
 		case *Call:
 			if f.Name == "top" || f.Name == "bottom" {
-				if len(f.Args) == 2 {
-					columnNames = append(columnNames, f.Name)
-					continue
+				for _, arg := range f.Args[1:] {
```

Comment:
I believe the number of args gets verified as part of the `validateAggregates` function so this shouldn't ever be zero.